lib/mix/lib/mix/tasks/test.ex

```diff
@@ -288,6 +290,7 @@ defmodule Mix.Tasks.Test do
     cover: :boolean,
     trace: :boolean,
     max_cases: :integer,
+    max_failures: :string,
```

Comment:
The `--slowest` flag uses `0` instead of `:infinity` to disable the feature, what do you think about taking the same approach here?

Reply:
@fertapric you have to remember that you will always have one more failure than max-failure.
so `--max-failure 0`  will stop on the first failure.
If we want something like that we should rename the feature to something like "stop-on-nth-failure" (I'm not really good at coming up with new names :-)

Reply:
agree with @fertapric, it makes more sense to have `--max-failures 1` -> stopping after first failure.

Reply:
I have changed the behaviour to stop when max-failures is reached. Though I haven't changed the default value being `:infinity` until we reach consensus.

Reply:
I thought about that too, and I also think `:infinity` is better for this feature. Using `0` with `--slowest` makes sense, but using it for disabling `--max-failures` would not be semantically correct. 

However, the behavior of `--max-failures 0` is a bit confusing. I guess I would expect to not run any tests 🤔

Reply:
I think we could raise if max-failures is 0, since it's probably an undesired behaviour